index.js

```diff
@@ -29,24 +29,35 @@ function dataUriToBuffer (uri) {
   // remove the "data:" scheme and parse the metadata
   var meta = uri.substring(5, firstComma).split(';');
 
+  var type = meta[0] || 'text/plain';
+  var typeFull = type;
   var base64 = false;
-  var charset = 'US-ASCII';
-  for (var i = 0; i < meta.length; i++) {
+  var charset = '';
+  for (var i = 1; i < meta.length; i++) {
     if ('base64' == meta[i]) {
       base64 = true;
-    } else if (0 == meta[i].indexOf('charset=')) {
-      charset = meta[i].substring(8);
+    } else {
+      typeFull += ';' + meta[i];
+      if (0 == meta[i].indexOf('charset=')) {
+        charset = meta[i].substring(8);
+      }
     }
   }
+  // defaults to US-ASCII only if type is not provided
```

Comment:
What's the thought on this API change? We're still defaulting the type to `text/plain` so why not have a default charset as well?